### ISSUE TITLE
Build an admin dashboard so that we can manage what goes on within the site

### ISSUE THREAD

**Issue by stemmlerjs** [stemmlerjs/ddd-forum#6]
We can build out an admin dashboard to manage what happens on the site:

As an admin user, I can:

- see the most recent comments
- delete comments that are inappropriate or useless
- ban users

etc... Open for suggestions :)

**Comment by lucianlature** [comment]
@stemmlerjs Any preference for the admin dashboard in terms of design system, visuals, stuff like that?...Or you just want to stick to the basics for now?

**Comment by stemmlerjs** [comment]
@lucianlature I think basic React components with minimal styling would be really good for an MVP here! Wanna start on this?

I'm starting to move towards using React + Styled Components, but that's probably unnecessary for now.

Think _1991 website_. We can improve the styling later. What we **do** need right up front is to know the use cases and make those work.

- Get recent comments
- Get popular comments
- Get users
- Get user by username or id
- Ban user
- Get posts
- Get post by post or slug id
- Delete post
- Get comments
- Get comments by post id
- Delete comment

Anything else?

**Comment by lucianlature** [comment]
@stemmlerjs 

1. Du you think [Material UI](https://material-ui.com/) is too much for the job?...
2. Outside the immediate scope of this discussion, any good resources on how to handle "many-to-many" relational mapping between entities in an aggregate root? 😄

**Comment by lucianlature** [comment]
> What we do need right up front is to know the use cases and make those work.

> * Get recent comments
> * Get popular comments
> * Get users
> * Get user by username or id
> * Ban user
> * Get posts
> * Get post by post or slug id
> * Delete post
> * Get comments
> * Get comments by post id
> * Delete comment

I'll build the basic layout for the admin dashboard, and then for each of the above use cases create PR(s). The next deployments should reflect the new functionality.

**Comment by stemmlerjs** [comment]
@lucianlature 

With respect to the UI, I'm gonna say yeah- go ahead with Material UI. It's actually pretty sweet. I really like the Firebase template :) looks cleaaaan. My primary concern was that if we choose something that not everyone is familiar with, it might make it harder for others to contribute, but I think what's much more interesting here is going to be the backend architecture and showing how we can design applications to serve multiple bounded contexts.

TLDR; do it 👍 :)  

> Outside the immediate scope of this discussion, any good resources on how to handle "many-to-many" relational mapping between entities in an aggregate root?

Want to open an issue for that [here](https://github.com/stemmlerjs/khalilstemmler-website-2019/pulls)? I'd love to chat about it. What's the domain that you're working in? (throw that in the issue as well :p).

> I'll build the basic layout for the admin dashboard, and then for each of the above use cases create PR(s). The next deployments should reflect the new functionality.

Excellent!! Thanks so much for your help on this. I won't get in your way too much for the UI side of things. Looking forward to it.

**Comment by Rammehar** [comment]
Hi, I have a question, should I create different api for admin login and end user login. I have different frontend for admin and end user. any help would be very much appreciated.

**Comment by gbrennon** [comment]
hey folks, how are u doing?

what is the current status of this issue? how can i help u 2?

**Comment by tdadadavid** [comment]
Hey guys, How is the issue going?
What is the current status has @gbrennon asked
I can help and contribute.

**Comment by gbrennon** [comment]
hey folks, how are u doing? do we have any status report for this issue?
